defs/go.bzl

```diff
@@ -62,15 +50,18 @@ def _compute_genrule_command(ctx):
   return '\n'.join(cmd)
 
 def _go_genrule_impl(ctx):
-  go_toolchain = get_go_toolchain(ctx)
-  all_srcs = set(go_toolchain.stdlib)
+  go_toolchain = ctx.toolchains["@io_bazel_rules_go//go:toolchain"]
+  all_srcs = depset(go_toolchain.data.stdlib)
   label_dict = {}
 
   for dep in ctx.attr.go_deps:
-    all_srcs = all_srcs | dep.transitive_sources
+    direct = dep[GoLibrary]
```

Comment:
Something about this confuses me. What is the type of `GoLibrary`?

Reply:
`GoLibrary` is a [provider](https://github.com/bazelbuild/rules_go/blob/fc20d84878ad8cd15bef0c7a0040cf2cf7c8ce4f/go/private/providers.bzl#L15).

This line is basically extracting [this struct](https://github.com/bazelbuild/rules_go/blob/fc20d84878ad8cd15bef0c7a0040cf2cf7c8ce4f/go/private/actions/library.bzl#L128-L138).